build-system/tasks/runtime-test/status-report.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const argv = require('minimist')(process.argv.slice(2));
+const log = require('fancy-log');
+const request = require('request');
+const {cyan, green, yellow} = require('ansi-colors');
+const {gitCommitHash} = require('../../git');
+const {isTravisPullRequestBuild} = require('../../travis');
+
+const reportBaseUrl = 'https://amp-test-status-bot.appspot.com/v0/tests/';
+
+function inferTestType() {
```

Comment:
Sorry - just catching up here. Is the idea to replace the `continuous-integration/travis-ci/pr` check? It's a bit confusing what the test types are because they're not a 1:1 match with the build jobs on Travis.

Reply:
It's not meant to replace Travis completely, but rather expose the status of each test type upwards. Eventually we want to change it so that test failures won't make the Travis build red (but other issues, such as Lint, will) and instead make the test status line red. Makes sense?

Reply:
Yeah it makes sense in that I get what this is trying to do, but I still think it's confusing for a developer to see both Travis and test checks on their PRs. If a test check fails, they'll have to do some extra thinking to see which job to look at or restart. (Also personally I think any test failure should make the Travis check red because that's what a CI build is expected to do).

I like the idea of bubbling up the results, but feel that checks should be 1:1 with Travis jobs to minimize confusion, and then remove the Travis check completely.

Reply:
One of the features I'll add soon is a link back to the Travis build.

The issue with leaving the Travis check red is that, since it's a required check, the build cop won't be able to merge PRs that are failing due to flaky tests/infrastructure, which is the whole intent of #20744. The tests have to run _somewhere_, so we can't remove the Travis check completely

Reply:
Hmm okay!